src/merge.ts

```diff
@@ -1,24 +1,114 @@
-import { IHttpHeaderParam, IHttpOperation, IMediaTypeContent, IServer } from '@stoplight/types';
+import {
+  IHttpOperation,
+  IHttpOperationRequestBody,
+  IHttpParam,
+  IMediaTypeContent,
+  INodeExample,
+  INodeExternalExample,
+  IServer,
+} from '@stoplight/types';
+import { JSONSchema4, JSONSchema6, JSONSchema7 } from 'json-schema';
+import { isEqual } from 'lodash';
 
-const mergeHeaders = mergeLists<IHttpHeaderParam[]>(
-  (h1, h2) => h1.name.toLowerCase() === h2.name.toLowerCase(),
-  h1 => h1, // ignore header #2 if mediaTypes is equal
-);
+type JSONSchema = JSONSchema4 | JSONSchema6 | JSONSchema7;
+
+function isSimpleAnyOfSchema(schema: JSONSchema) {
+  return schema.anyOf && Object.keys(schema).length === 1;
+}
+
+function mergeSchemas(schema1: JSONSchema, schema2: JSONSchema): JSONSchema {
+  const schemas = (isSimpleAnyOfSchema(schema2) ? (schema2.anyOf as JSONSchema[]) : [schema2]).reduce<
+    Array<JSONSchema>
+  >(
+    (schemas, schema) => {
+      if (!schemas.find(s => isEqual(s, schema))) {
+        schemas.push(schema);
+      }
+      return schemas;
+    },
+    isSimpleAnyOfSchema(schema1) ? (schema1.anyOf as JSONSchema[]) : [schema1],
```

Comment:
If we use the type guard above, the cast here is not required anymore
```suggestion
    isSimpleAnyOfSchema(schema1) ? schema1.anyOf : [schema1],
```

Reply:
I added the type guard which casts to `JSONSchema7`. The `JSONSchema` types compatibility is a bit problematic between 4,6,7 versions. They differ at some places and I often just get a huge typing error from the compiler.

The current state works. I don't know why.